honeybee_radiance_command/gensky.py

```diff
@@ -0,0 +1,228 @@
+"""gensky command."""
+
+from .options.gensky import GenskyOptions
+from ._command import Command
+import honeybee_radiance_command._exception as exceptions
+
+
+class Gensky(Command):
+    """gensky command."""
+
+    __slots__ = ('_month', '_day', '_time', '_time_zone', '_solar_time', '_input')
+
+    def __init__(self, month, day, time, time_zone=None, solar_time=False,
+                 options=None, output=None):
+        """Command.
+
+        Args:
+            month: An integer representing the number of the month. Count starts from 01.
+            day: An integer representing the number of the day in a month. Count starts
+                from 01.
+            time: A float representing hour and minute.
+            time_zone: A three letter text representing the time zone.
+                Following are acceptable time zones with their corresponding hour
+                differences from Greenwhich mean time;
+                YST +9, PST +8, MST +7, CST +6, EST +5, GMT 0, CET -1, EET -2, AST -3,
+                GST -4, IST -5.5, JST -9, NZST -12, YDT +8, PDT +7, MDT +6, CDT +5,
+                EDT +4, BST -1, CEST -2, EEST -3, ADT -4, GDT -5, IDT -6.5, JDT -10,
+                NZDT -13
+            solar_time: A boolean to use local solar time
+            options: Command options. It will be set to Radiance default values if not
+                provided by user.
+            output: File path to the output file (Default: None).
+        """
+        Command.__init__(self, output=output)
+        if options:
+            self._options = options
+        else:
+            self._options = GenskyOptions()
+
+        self._month = self.check_month(month)
+        self._day = self.check_day(day)
+        self._time = self.check_time(time)
+        self._time_zone = self.check_time_zone(time_zone)
+        self._solar_time = self.check_solar_time(solar_time)
+
+    @staticmethod
+    def check_month(month):
+        """Validate the value of month."""
+        if month and isinstance(month, int) and month in range(1, 13):
+            return month
+        else:
+            raise ValueError('The value of month needs to be an integer between'
+                             ' 1 and 12 inclusive. Instead got %.' % (month))
+
+    @staticmethod
+    def check_day(day):
+        """Validate the value of day."""
+        if day and isinstance(day, int) and day in range(1, 32):
+            return day
+        else:
+            raise ValueError('The value of day needs to be an integer between'
+                             ' 1 and 31 inclusive. Instead got %.' % (day))
+
+    @staticmethod
+    def check_time(time):
+        """Validate the value of time."""
+        if time and isinstance(time, float):
+            hour, minute = str(time).split('.')
+            if int(hour) in range(1, 25) and int(minute) in range(0, 60):
```

Comment:
Why not doing it the easy way? You are generating a range each time with this design.

```suggestion
            if 1 <= int(hour) <= 24 and 0 <= int(minute) <= 59:
```